tensorflow_datasets/core/download/kaggle.py

```diff
@@ -46,10 +47,44 @@
 class KaggleFile(object):
   """Represents a Kaggle competition file."""
   _URL_PREFIX = "kaggle://"
+  kaggle_type = collections.namedtuple(
+      "KaggleType",
+      ["prefix", "download_cmd", "dl_flag", "extra_flag"])
+  _DATASET_TYPE = kaggle_type(
+      prefix="dataset",
+      download_cmd="datasets",
+      dl_flag="-d",
+      extra_flag="--unzip")
+  _COMPETITION_TYPE = kaggle_type(
+      prefix="competition",
+      download_cmd="competitions",
+      dl_flag="-c",
+      extra_flag="")
+  KAGGLE_TYPES = {
+      _COMPETITION_TYPE.prefix: _COMPETITION_TYPE,
+      _DATASET_TYPE.prefix: _DATASET_TYPE
+      }
 
   def __init__(self, competition_name, filename):
     self._competition_name = competition_name
     self._filename = filename
+    self._type = None
```

Comment:
Why not directly assign the type here and remove the property ?

Reply:
Well, I created a blank variable here because I was trying to avoid `KaggleFile.get_type()` to be called everytime one accesses the `type` property. Essentially determining once(when `self._type` is `None`) and storing the result so that one doesn't need to call the static function for further calls to the property.

Reply:
I meant why not have

```
def __init__():
  self._type = get_kaggle_type()
```

And remove the property entirely.

Reply:
Done :)